File: Python/BOJ/덱/5430.py
import sys

t = int(sys.stdin.readline().rstrip())
for _ in range(t):
  commands = list(sys.stdin.readline().rstrip())
  n = int(sys.stdin.readline().rstrip())
  inputs = sys.stdin.readline().rstrip()
  if inputs == "[]":
    arr = []
  else:
    arr = list(map(int, inputs[1:-1].split(sep= ",")))

  head = 0; tail = len(arr) - 1
  isReverse = False
  isError = False
  for cmd in commands:
    if cmd == "R":
      isReverse = not isReverse
    elif cmd == "D":
      if head > tail:
        isError = True
        break
      else:
        if isReverse: tail -= 1
        else: head += 1

  if isError:
    print("error")
  elif head > tail:
    print("[]")
  else:
    result = arr[head:tail+1]
    if isReverse:
      result.reverse()
    print("[", end= "")
    for num in result[0:-1]:
      print(num, end= ",")
    print(result[-1], end= "")
    print("]")

File: Swift/BOJ/문제집/스택/5430.swift
//
//  5430.swift
//  Algorithm
//
//  Created by 유정주 on 2023/07/03.
//

import Foundation

let t = Int(readLine()!)!
for _ in 0..<t {
    let commands = Array(readLine()!)
    let n = Int(readLine()!)!
    let arr = readLine()!.dropFirst().dropLast().split { $0 == "," }.map { Int($0)! }
    
    var head = 0, tail = arr.count - 1
    var isReversed = false
    var isError = false
    for command in commands {
        if command == "R" {
            isReversed.toggle()
        } else if command == "D" {
            if head > tail {
                isError = true
                break
            }
            
            if isReversed {
                tail -= 1
            } else {
                head += 1
            }
        }
    }
    
    if isError {
        print("error")
    } else if head > tail {
        print("[]")
    } else {
        if isReversed {
            print("[\(arr[head...tail].reversed().map { String($0) }.joined(separator: ","))]")
        } else {
            print("[\(arr[head...tail].map { String($0) }.joined(separator: ","))]")
        }
        
    }
}
